boards/nRF52840-DK/src/lib.rs

```diff
@@ -0,0 +1,580 @@
+//! Board support crate for the Nordic nRF52840-DK
+//!
+//! This crate is in early development. Not much to see here, right now.
+#![no_std]
+
+//#![deny(missing_docs)]
+#![deny(warnings)]
+
+pub use cortex_m;
+pub use cortex_m_rt;
+pub use embedded_hal;
+pub use nrf52840_hal as hal;
+
+/// Exports traits that are usually needed when using this crate
+pub mod prelude {
+    pub use nrf52840_hal::prelude::*;
+}
+
+// TODO:
+// pub mod debug;
+
+use nrf52840_hal::{
+    prelude::*,
+    gpio::{
+        p0::{
+            self,
+        },
+        p1,
+        Pin,
+        Floating,
+        Input,
+        Output,
+        PushPull,
+        PullUp,
+        Level,
+    },
+    nrf52840_pac::{
+        self as nrf52,
+        CorePeripherals,
+        Peripherals,
+    },
+    spim::{
+        self,
+        Spim
+    },
+    uarte::{
+        self,
+        Uarte,
+        Parity as UartParity,
+        Baudrate as UartBaudrate,
+    },
+};
+
+/// Provides access to all features of the nRF52840-DK board
+#[allow(non_snake_case)]
+#[allow(non_camel_case_types)]
+pub struct nRF52840DK {
+    /// The nRF52's pins which are not otherwise occupied on the nRF52840-DK
+    pub pins: Pins,
+
+    /// The nRF52840-DK UART which is wired to the virtual COM port
+    pub com: Uarte<nrf52::UARTE0>,
+
+    /// The nRF52840-DK SPI which is wired to the SPI flash
+    pub flash: Spim<nrf52::SPIM2>,
+    pub flash_cs: Pin<Output<PushPull>>,
+
+    /// The LEDs on the nRF52840-DK board
+    pub leds: Leds,
+
+    /// The buttons on the nRF52840-DK board
+    pub buttons: Buttons,
+
+    pub nfc: NFC,
+
+    /// Core peripheral: Cache and branch predictor maintenance operations
+    pub CBP: nrf52::CBP,
+
+    /// Core peripheral: CPUID
+    pub CPUID: nrf52::CPUID,
+
+    /// Core peripheral: Debug Control Block
+    pub DCB: nrf52::DCB,
+
+    /// Core peripheral: Data Watchpoint and Trace unit
+    pub DWT: nrf52::DWT,
+
+    /// Core peripheral: Flash Patch and Breakpoint unit
+    pub FPB: nrf52::FPB,
+
+    /// Core peripheral: Floating Point Unit
+    pub FPU: nrf52::FPU,
+
+    /// Core peripheral: Instrumentation Trace Macrocell
+    pub ITM: nrf52::ITM,
+
+    /// Core peripheral: Memory Protection Unit
+    pub MPU: nrf52::MPU,
+
+    /// Core peripheral: Nested Vector Interrupt Controller
+    pub NVIC: nrf52::NVIC,
+
+    /// Core peripheral: System Control Block
+    pub SCB: nrf52::SCB,
+
+    /// Core peripheral: SysTick Timer
+    pub SYST: nrf52::SYST,
+
+    /// Core peripheral: Trace Port Interface Unit
+    pub TPIU: nrf52::TPIU,
+
+    /// nRF52 peripheral: FICR
+    pub FICR: nrf52::FICR,
+
+    /// nRF52 peripheral: UICR
+    pub UICR: nrf52::UICR,
+
+    // /// nRF52 peripheral: BPROT
+    // TODO:
```

Comment:
TODO?

Reply:
I left this, but elaborated on why it is there :)